src/Controller/EloquaAppCloudEndpointController.php

```diff
@@ -0,0 +1,303 @@
+   * @param \Drupal\Core\Queue\QueueFactory $queue
+   * @param $plugins
+   */
+  public function __construct(ClientFactory $eloquaFactory, RequestStack $requestStack, LanguageManagerInterface $langManager, EntityTypeManager $entityManager, QueueFactory $queueFactory,array $plugins) {
+    $this->eloqua = $eloquaFactory->get();
+    $this->request = $requestStack->getCurrentRequest();
+    $this->langManager = $langManager;
+    $this->entityManager = $entityManager;
+    $this->queueFactory = $queueFactory;
+    $this->plugins = $plugins;
+  }
+
+  /**
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+
+    // Get the list of plugin managers in our "namespace".
+    foreach ($container->getServiceIds() as $serviceId) {
+      if (strpos($serviceId, 'plugin.manager.eloqua_app_cloud') === 0) {
+        $type = $container->get($serviceId);
+        $plugin_definitions = $type->getDefinitions();
+
+        foreach ($plugin_definitions as $plugin) {
+          $plugins[$plugin['id']] = $type;
+        }
+      }
+    }
+
+    return new static(
+      $container->get('eloqua.client_factory'),
+      $container->get('request_stack'),
+      $container->get('language_manager'),
+      $container->get('entity_type.manager'),
+      $container->get('queue'),
+      $plugins
+    );
+  }
+
+  /**
+   * Instantiate and store the instance ID from Eloqua.
+   *
+   * @return \Drupal\eloqua_app_cloud\Controller\JsonResponse
+   */
+  public function instantiate($eloqua_app_cloud_service) {
+    // Get the instanceID from the query parameter.
+    $instanceId = $this->request->get("instance");
+    if (empty($instanceId)) {
+      //Throw an exception, and/or return an error?
+    }
+    $pluginReferences = $this->getEntityPlugins($eloqua_app_cloud_service);
+
+    // Iterate over the ServiceEntity plugins and build a merged field list.
+    foreach ($pluginReferences as $pluginReference) {
+      $id = $pluginReference->value;
+      // Get the plugin manager from the list of plugins (from the container).
+      $pluginMgr = $this->plugins[$id];
+      // Instantiate the referenced plugin.
+      $plugin = $pluginMgr->createInstance($id);
+      // We can only allow ONE type of API at a time.
+      if(!empty($api) && $api !== $plugin->api()){
+        // @TODO Throw an exception!
+      }
+      $api = $plugin->api();
+      // @TODO figure out how to determine field list? This pulls it from annotation, but maybe it should be in config so it can be customized?
+      $fieldList = $plugin->fieldList();
+      $fieldList = array_merge($fieldList, $plugin->fieldList());
+    }
+
+    $response = new \stdClass();
+    $response->recordDefinition = $fieldList;
+    $response->requiresConfiguration = FALSE;
+    \Drupal::logger('EloquaAppCloudDecision')->notice(print_r($response, TRUE));
```

Comment:
- [ ] I would recommend injecting a logger with this module's channel, here (and throughout).  https://www.drupal.org/docs/8/api/logging-api/overview

For future unit testability sanity.

Reply:
Maybe also some more helpful verbiage around the message itself, using the string formatting stuff, etc.

Reply:
That one was a debugging step I forgot to delete.